plugins/woocommerce-admin/client/marketplace/components/feedback-modal/feedback-modal.tsx

```diff
@@ -0,0 +1,152 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Modal, Button, TextControl, TextareaControl } from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
+import { recordEvent } from '@woocommerce/tracks';
+
+/**
+ * Internal dependencies
+ */
+import './feedback-modal.scss';
+import LikertScale from '../likert-scale/likert-scale';
+
+export default function FeedbackModal(): JSX.Element {
+	const CUSTOMER_EFFORT_SCORE_ACTION = 'marketplace_redesign_2023';
+	const LOCALSTORAGE_KEY_DISMISSAL_COUNT = 'marketplace_redesign_2023_dismissals'; // ensure we don't ask for feedback if the user's already given feedback or declined to multiple times
+	const LOCALSTORAGE_KEY_LAST_REQUESTED_DATE = 'marketplace_redesign_2023_last_shown_date'; // ensure we don't ask for feedback more than once per day
+	const SUPPRESS_IF_DISMISSED_X_TIMES = 2; // if the user dismisses the snackbar this many times, stop asking for feedback
+	const SUPPRESS_IF_AFTER_DATE = '2024-01-01'; // if this date is reached, stop asking for feedback
```

Comment:
🗒️  **For discussion!** Is "2 dismissals then it goes away forever" enough? I think so, but I'm open to other thoughts. Is the "suppress after date" right? Open to opinions.

Reply:
I would say 1 dismisal is enough to make it go away. I'm so fed up of being asked for feedback everywhere I go on the interwebs!

`SUPPRESS_IF_AFTER_DATE` seems fair to me.

Reply:
I agree with Gareth. I understand the idea that we sometimes dismiss these things temporarily and so in that case we may miss feedback, but doing it too many times is just annoying. I'd be happy for 1, but I think 2 may be better in this case.